### DISCUSSION TITLE
Future of lntop

### DISCUSSION THREAD

**Discussion by edouardparis** [edouardparis/lntop#83] [General]
Current situation:

- @edouardparis is not using lnd anymore for his personal use. 
- Code reviews takes time and commitment and sadly PRs with very valuable fix and features are stacking without being merged

Should we:
 - merge after review all current request and do a new release.
 - then archive the repository and give a link to forward user to an other fork ?

**Comment by edouardparis** [comment]
ping @rkfg and @hieblmi

**Comment by rkfg** [comment]
Just out of curiosity, did you switch to another LN implementation or moved away from LN/Bitcoin completely? In the former case I think the architecture anticipates multiple supported backends even though only lnd is implemented.

Github supports renaming repositories so that the old links aren't broken and simply redirect to the new location. If you don't want to be strictly associated with lntop you can create an organization (`github.com/lntop` seems to be available?) and move the project there. Then you can give people access to commit, merge and such. I use lntop daily and write in Go professionally at work so I intend to keep fixing various pain points I encounter.

**Reply by edouardparis** [reply]
I am currently working for https://github.com/revault/ for future "onchain" products. I am still following lightning current development but I am not invested in lightning app or routing node management anymore.

The multiple backend support was overly optimistic that the different implementations of lightning will have similarities in their API like a simple command to list the channels of the node. I never tried to dig the details and I am thinking now that lntop should maybe support lnd only.

For the future:
I dont want my lack of engagement to be an obstacle for the project and repel new contributors.
I still can keep an eyes on the advancement of the project and help sometimes.

A new organization is good I guess or maybe @lightninglabs would be interested to host it ?

**Reply by rkfg** [reply]
Thanks for the clarifications and best of luck for your future endeavors!

If we leave lnd only it can be greatly simplified and refactored by throwing away many structures and proto converters since we'd be able to use `lnrpc.*` structures directly. Not a small task but doable.

Would be awesome to find a new home for lntop.

**Reply by hieblmi** [reply]
Along the lines of @rkfg I am using `lntop` daily and am very much interested in seeing new features being made available in a reasonable time. I am also planning to contribute more to this repo as in as much as my schedule allows so it would be great if we could lift this repo to a public space and distribute merge access. Would be amazing if @lightninglabs would offer to host it.

**Reply by hieblmi** [reply]
@edouardparis @rkfg I checked with someone from lightning labs and they said they currently don't have capacity to host/support the `lntop` repo. So I am checking in with you to see what our other options are to continue dev/maintenance of `lntop`.

I could fork the repo and give access to @rkfg if there's still interest. Do you have other suggestions?

**Reply by rkfg** [reply]
Well, if it needs a fork I can do it myself easily. Currently I'm busy with other things though so there's no need.

**Reply by edouardparis** [reply]
Thanks for asking them.
I am gonna merge @hieblmi last PR and release one last binary, then If one of you want to do a fork, I will archive the repo and put a link to it.

**Reply by HappyBear-btc** [reply]
I really like lntop. It's simple and shows the data live. Hope it will be developed further

**Reply by hieblmi** [reply]
@HappyBear-btc which features would you like to see?

**Reply by HappyBear-btc** [reply]
It's actually so perfect tool! 
I always have it open as live monitoring, it's cool and informative just the best overview that you can have online from the terminal.
I guess there're always requests for features and it would be a shame if lntop is abandoned.
so in general I hope lntop stays alive in terms of maintenance and development. 
btw for new users it would be convenient to add "a/d-to sort column" in the bottom near F2-Menu,Enter-details,F10-Quit.

**Reply by HappyBear-btc** [reply]
@hieblmi I noticed that Forward history is not a live screen, it only updates after you switch to other screen and cone back.
would be useful to make it live like the other screens.
